src/collections/Menu/MenuItem.js

```diff
@@ -8,16 +8,21 @@ import {
 } from '../../lib'
 
 function MenuItem(props) {
-  const { active, children, className, header } = props
+  const { active, children, className, header, href, link, onClick } = props
   const classes = cx(
     className,
     useKeyOnly(active, 'active'),
     useKeyOnly(header, 'header'),
+    useKeyOnly(link, 'link'),
     'item',
   )
+  const handleClick = (e) => {
+    if (onClick) onClick(e)
+  }
   const rest = getUnhandledProps(MenuItem, props)
+  const ItemComponent = href || onClick ? 'a' : 'div'
 
-  return <a {...rest} className={classes}>{children}</a>
+  return <ItemComponent {...rest} className={classes} onClick={handleClick}>{children}</ItemComponent>
```

Comment:
This is not the first issue we've had with rendering `a` tags when there is an onClick prop.  I'm starting to think we only do so when there is an `href` or some explicit `link` prop.

Also, many other components and use cases have us considering the use of a library wide `as` prop:

```jsx
<Segment as='section' />
//=> <section class="ui segment"></section>

<MenuItem as='a' />
//=> <a class="item></a>

<Dropdown as={Button} options={[...]} />
//=> <div class="ui button dropdown" />
```

Especially handy for `react-router` use with the MenuItem as it would allow passing props for both components:

```jsx
import { Link } from 'react-router'

<MenuItem link as={Link} to='/home' activeClass='active' />
//=> <a class="item active"></a>
```

The `MenuItem` would consume its own props (`link`) and spread the `rest`.  The `as` prop here means the MenuItem would render like:

```jsx
render() {
  const ElementType = getElementType(props) // new util, like getUnhandledProps()

  return <ElementType {...rest} /* other menu item props */ />
}
```

I'm thinking of doing this today.  The `getElementType()` method would hold the logic for returning the value of the `as` prop, else if props.link return `a`, else `div`.  We could add more logic to how components are allowed to determine their element type as well.  This is good because all components would be consistent.

Feedback welcome here #403 

***

For now, just render a `div` unless there is a `link` or `href` prop.